cilium-cli/connectivity/tests/dns.go

```diff
@@ -0,0 +1,80 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package tests
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/cilium/cilium/cilium-cli/connectivity/check"
+	"github.com/cilium/cilium/cilium-cli/utils/features"
+)
+
+// PodToExternalDNS is a test scenario which tests DNS and DNSSEC connectivity from client pods to an
+// external DNS server.
+func PodToExternalDNS(targetName string) check.Scenario {
+	return &podToExternalDNS{
+		ScenarioBase: check.NewScenarioBase(),
+		targetName:   targetName,
+	}
+}
+
+// podToExternalDNS implements a Scenario.
+type podToExternalDNS struct {
+	check.ScenarioBase
+	targetName string
+}
+
+func (p *podToExternalDNS) Name() string {
+	return fmt.Sprintf("pod-to-external-dns-%s", p.targetName)
+}
+
+func (p *podToExternalDNS) Run(ctx context.Context, t *check.Test) {
+	ct := t.Context()
+
+	dnsServer := check.DNSEndpoint(
+		fmt.Sprintf("dns-server-%s", ct.Params().ExternalDNSServerIP),
+		ct.Params().ExternalDNSServerIP,
+		uint32(ct.Params().ExternalDNSServerPort),
+	)
+
+	target := check.DNSEndpoint(fmt.Sprintf("dns-name-%s", p.targetName), p.targetName, 0)
+
+	i := 0
+	for _, client := range t.Context().ClientPods() {
+		t.ForEachIPFamily(func(ipFam features.IPFamily) {
+			// TODO: Reenable the test once the kernel with the bugfix is released:
+			// https://patchwork.kernel.org/project/netdevbpf/patch/20250318161516.3791383-1-maxim@isovalent.com/
+			// and when IPv6 external connectivity starts working in the CI.
+			if ipFam == features.IPFamilyV6 {
+				return
+			}
+
+			// DNS query (pod to external DNS)
+			t.NewAction(p, fmt.Sprintf("dig-%d", i), &client, target, ipFam).Run(func(a *check.Action) {
+				a.ExecInPod(ctx, ct.DigCommand(dnsServer, ipFam, p.targetName))
+			})
+		})
+
+		i++
+	}
+
+	i = 0
+	for _, client := range t.Context().ClientPods() {
+		t.ForEachIPFamily(func(ipFam features.IPFamily) {
+			// TODO: Reenable the test once the kernel with the bugfix is released:
+			// https://patchwork.kernel.org/project/netdevbpf/patch/20250318161516.3791383-1-maxim@isovalent.com/
+			// and when IPv6 external connectivity starts working in the CI.
+			if ipFam == features.IPFamilyV6 {
+				return
+			}
+
+			// DNSSEC query (pod to external DNS)
+			t.NewAction(p, fmt.Sprintf("dig-dnssec-%d", i), &client, target, ipFam).Run(func(a *check.Action) {
+				a.ExecInPod(ctx, ct.DigCommand(dnsServer, ipFam, p.targetName, "+dnssec"))
```

Comment:
I'm not sure I understand - does `dig +dnssec` fail if the `targetName` domain has no DNSSEC enabled? Does `www.cilium.io` have DNSSEC enabled?

Reply:
It checks for a DNSSEC record for `targetName`, so yes it fails if DNSSEC is not enabled for the given domain. That's what the existing runtime FQDN test currently also does.